woocommerce/woocommerce-android
# AI: Generate product sharing message part 2: Functionality

Please do not merge before #9224 is merged.

Part of: https://github.com/woocommerce/woocommerce-android/issues/9216
Ref: pe5sF9-1AH-p2

Sorry if the PR is a bit large; a good chunk of it is just the Prompts text, and trackings.

### Description


This PR continues from #9224 by adding the actual functionalities to the UI.

The added features are:
1. "Share" button now works, with or without sharing message.
2. Generating share message with AI now works,
3. Information button now opens to Automattic's AI guidelines page.
4. Error handling is now added in case AI completion fetching fails.
5. Analytics:

Event | Trigger | Properties
-- | -- | --
*_product_sharing_ai_displayed | When the sheet for AI-generated share content is displayed. |  
*_product_sharing_ai_generate_tapped | When the “Write with AI” button is tapped on the AI product share sheet. | is_retry: true \| false
*_product_sharing_ai_share_tapped | When the “Share” button is tapped on the AI product share sheet. | with_message: true \| false
*_product_sharing_ai_dismissed | When the AI product share sheet is dismissed. |  
*_product_sharing_ai_message_generated | When a message is generated on the AI product share sheet. |  
*_product_sharing_ai_message_generation_failed | When message generation fails on the AI product share sheet. | error_context: classname error_type: errorType error_description: errorDescription

### Testing instructions


1. Start app, go to Products, select a published Product,
2. tap Share on option on top right, then the sharing dialog will show up,
3. Test sharing without a message,
4. Go back, test the Write with AI button and make sure a message is generated,
5. Test the "Regenerate" button,
7. Test sharing with your own manually written message,
8. Test the (i) icon and make sure it opens the webview to Automattic's AI Guidelines page.
9. Open share dialog, turn off internet, and check the error case.
10. Lastly, check logcat and find matching tracking events as outlined above.

Check video below for testing example:

### Video


[Demo for product sharing with AI](https://github.com/woocommerce/woocommerce-android/assets/266376/75a48126-0a5c-42d6-9a66-0ad404a76966)
